utils/exceptions.py

```diff
@@ -0,0 +1,84 @@
+from typing import Optional
+
+from django.http import Http404
+from rest_framework import status
+from rest_framework.exceptions import APIException, ValidationError
+from rest_framework.response import Response
+
+
+def _get_error_wrapper(data) -> dict[str, list]:
+    if not isinstance(data, list):
+        data = [data]
+
+    return {"errors": data}
+
+
+def _get_error_detail(code, detail) -> dict[str, list]:
+    return {"code": code.upper(), "message": detail}
+
+
+def get_error_response(code, detail) -> dict[str, list]:
+    return _get_error_wrapper(_get_error_detail(code.upper(), detail))
+
+
+def sentry_before_send(event, hint):
+    """
+    Filter internal exceptions that should not be logged to Sentry
+    https://docs.sentry.io/platforms/python/guides/django/configuration/filtering/
+    """
+    from atv.exceptions import ATVError
+
+    if "exc_info" in hint:
+        exc_type, exc_value, tb = hint["exc_info"]
+        if isinstance(exc_value, ATVError):
+            return None
+    return event
+
+
+def custom_exception_handler(exc, _context=None) -> Optional[Response]:
```

Comment:
The default exception handler does sometimes return `None`, but to me looks like that this function will always return a response?

Reply:
> _If the handler returns None then the exception will be re-raised and Django will return a standard HTTP 500 'server error' response._

In this case, to have a consistent error response, instead of letting Django handle the "standard HTTP 500" error, we return our own error with a proper error message.

Reply:
Indeed. I just meant that maybe `-> Optional[Response]` should be `-> Response`

Reply:
Ahhh my bad 😅 yeah, I'll fix that

Reply:
Done